federation/cmd/federated-apiserver/app/server.go

```diff
@@ -390,9 +240,8 @@ func Run(s *options.APIServer) error {
 
 	var serviceAccountGetter serviceaccount.ServiceAccountTokenGetter
 	if s.ServiceAccountLookup {
-		// If we need to look up service accounts and tokens,
-		// go directly to etcd to avoid recursive auth insanity
-		serviceAccountGetter = serviceaccountcontroller.NewGetterFromStorageInterface(etcdStorage)
+		// TODO: Maybe do not expose this option in genericapiserver, if most servers do not need it?
```

Comment:
The generic apiserver won't (necessarily) have direct access to the same etcd that the main apiserver uses. Also, the generic apiserver won't itself host the token or service account resources. Therefore, there's no question of having an infinite resource check loop, if I understand correctly what this is doing. The "delete token" operation would only be sent to the main apiserver.